include/marl/scheduler.h

```diff
@@ -147,6 +147,39 @@ class Scheduler {
                      const std::chrono::time_point<Clock, Duration>& timeout,
                      const Predicate& pred);
 
+    // wait() suspends execution of this Fiber until the Fiber is woken up with
+    // a call to notify().
+    // While the Fiber is suspended, the scheduler thread may continue executing
+    // other tasks.
+    // wait() must only be called on the currently executing fiber.
+    //
+    // Warning: Unlike wait() overloads that take a lock and predicate, this
+    // form of wait() offers no safety for notify() signals that occur before
+    // the fiber is suspended, when signalling between different threads. In
+    // this scenario you may deadlock. For this reason, it is only ever
+    // recommended to use this overload if you can guarantee that the calls to
+    // wait() and notify() are made by the same thread.
+    //
+    // Use with extreme caution.
+    inline void wait();
```

Comment:
I do wonder if perhaps wait_nolock() would be a good name for these functions? Like using cast operators, I think it's good to have a name for these specific waits, as you can search for them in code, and pay special attention when reading code that makes use of them. Alternative names: wait_unsafe, wait_same_thread, wait_lockless (though, "lockless" has special meaning and is no good, I think).

Reply:
I'm not massively keen on making the API smell for functions that (while are potentially error-prone) have their behaviours and warnings clearly documented.
This API is as low as you can get in marl, and if you find yourself using this instead of the numerous synchronisation primitives that cover 99% what you want, I'd like to hope you know what you are doing. `wait()` is already overloaded, it feels odd to me that this would get a special name.

Reply:
I think what bothers me most is that the dangerous wait() function is the one that takes no arguments. If you don't like renaming the function, perhaps passing it a tag such as 'enum {NoLock}' would make it less likely to be misused?

Reply:
It's still code smell. If you cannot be bothered to read the documentation on the functions you're calling here, you're going to have a bad time with any of these methods.